### PR TITLE
docker-machine-driver-hyperkit v1.0.0 (new formula)

### PR DESCRIPTION
**Pull request by jacobwgillespie**
This adds a formula for [`docker-machine-driver-hyperkit`](https://github.com/machine-drivers/docker-machine-driver-hyperkit).

---

- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

### CODE REVIEW

**Comment by jason-yoo-hs** [comment]
Could someone approve this PR soon? It would make my life a lot easier :)

**Comment by froch** [comment]
i'd love to use this instead of `minikube config set WantShowDriverDeprecationNotification false` after installing xhyve. please. thank you

**Comment by bric3** [comment]
Why not using the pre-built binary ? available here :

https://storage.googleapis.com/minikube/releases/v0.27.0/docker-machine-driver-hyperkit

or here for the _latest_ version

https://storage.googleapis.com/minikube/releases/latest/docker-machine-driver-hyperkit

Instructions are defined [here](https://github.com/kubernetes/minikube/blob/master/docs/drivers.md#hyperkit-driver)

In such case maybe the homepage should be targeting [this page](https://github.com/kubernetes/minikube/blob/master/docs/drivers.md#hyperkit-driver) as well ?

**Comment by externl** [comment]
@bric3  See https://docs.brew.sh/FAQ#why-do-you-compile-everything

**Comment by bric3** [comment]
@externl Ok, got it thanks for faq link ;)

File: Formula/docker-machine-driver-hyperkit.rb [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    :tag => "v0.27.0",
+    :revision => "a7c0cbee11dd97f257cc8d050605684c0c446a7e"
+
+  depends_on "go" => :build
+  depends_on :macos => :yosemite
+  depends_on "docker-machine" => :recommended
+
+  def install
+    ENV["GOPATH"] = buildpath
+    ENV["CGO_ENABLED"] = "1"
```
Comment (DomT4) [inline Formula/docker-machine-driver-hyperkit.rb]: This is the default; doesn't need writing.

Comment (jacobwgillespie) [reply Formula/docker-machine-driver-hyperkit.rb]: Thanks, updated. 👍

**Review by GabLeRoux — APPROVED** [review]

**Comment by jacobwgillespie** [comment]
Looks like the new formula audit now forbids recommended dependencies.  Updated.

**Comment by artheus** [comment]
It seems that this does not work with the default boot2docker.iso image this driver checks for the Kernel file bzimage which apparently does not exist in boot2docker.iso. It does not work to add the minikube.iso to the docker machine cache directory either, as it check that for the version and it will not be the correct one. I think that this should rather be created as a separate docker-machine-driver in a separate project, as this version is very specific to minikube.

**Comment by jacobwgillespie** [comment]
I have updated this formula to use [the new standalone repository](https://github.com/machine-drivers/docker-machine-driver-hyperkit) for `docker-machine-driver-hyperkit`.  It has been tagged there as `v1.0.0`.

@artheus, it looks like after updating this formula to v1.0.0, it no longer has a dependency on `bzimage` and appears to run successfully now with vanilla `boot2docker.iso`.

**Comment by jacobwgillespie** [comment]
> GitHub repository not notable enough (<30 forks, <30 watchers and <75 stars)

That's because this project was just extracted from its previous home (https://github.com/kubernetes/minikube, 1400 forks, 357 watchers, 10k stars).

**Comment by artheus** [comment]
Another alternative is to use this https://github.com/machine-drivers/docker-machine-driver-xhyve
Which implements it very well and give much more options than the minikube one. (From what I know)
It already exists in brew, and is installable using `brew install docker-machine-driver-xhyve`

Just create a symlink `ln -s /usr/local/bin/hyperkit /usr/local/bin/xhyve` and it works great!

**Comment by jacobwgillespie** [comment]
To quote from the hyperkit driver:

> The Hyperkit driver will eventually replace the existing xhyve driver.

The xhyve driver does already exist in Homebrew yes, but this formula is for anybody who would rather use the "official" hyperkit driver.

**Comment by joaosa** [comment]
I would love for this get to merged

**Comment by GabLeRoux** [comment]
In case someone doesn't know yet, we can already install this from formula url:

```bash
FORMULA_URL=https://raw.githubusercontent.com/jacobwgillespie/homebrew-core/docker-machine-driver-hyperkit/Formula/docker-machine-driver-hyperkit.rb
brew install $FORMULA_URL
```

I [tested this on travis](https://github.com/GabLeRoux/minikube-macos-hyperkit-instructions) before running on my system just in case. I confirm it works fine on my side 👍

File: Formula/docker-machine-driver-hyperkit.rb [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+      system "go", "build", "-o", "#{bin}/docker-machine-driver-hyperkit",
+             "-ldflags", "-X main.version=#{version}"
+      prefix.install_metafiles
+    end
+  end
+
+  def caveats; <<~EOS
+    This driver requires superuser privileges to access the hypervisor. To
+    enable, execute
+        sudo chown root:wheel #{opt_prefix}/bin/docker-machine-driver-hyperkit
```
Comment (DomT4) [inline Formula/docker-machine-driver-hyperkit.rb]: `opt_bin`. (I appreciate the other formula to do this has the wrong style used as well).

File: Formula/docker-machine-driver-hyperkit.rb [hunk 3]
```diff
@@ -0,0 +1,35 @@
+class DockerMachineDriverHyperkit < Formula
+  desc "Docker Machine driver for hyperkit"
+  homepage "https://github.com/machine-drivers/docker-machine-driver-hyperkit"
+  url "https://github.com/machine-drivers/docker-machine-driver-hyperkit.git",
+    :tag => "v1.0.0",
```
Comment (DomT4) [inline Formula/docker-machine-driver-hyperkit.rb]: Align this and `:revision` with the first `"` above, thanks!

File: Formula/docker-machine-driver-hyperkit.rb (hunk 2 above)
Comment (jacobwgillespie) [reply Formula/docker-machine-driver-hyperkit.rb]: Nice, fixed 👍

File: Formula/docker-machine-driver-hyperkit.rb (hunk 3 above)
Comment (jacobwgillespie) [reply Formula/docker-machine-driver-hyperkit.rb]: Updated 👍

**Comment by DomT4** [comment]
Couple more things the audit has picked up on (Who needs humans eh)

```
        docker-machine-driver-hyperkit:
  * C: 16: col 7: use "dep", "ensure", "-vendor-only"
  * C: 28: col 1: `EOS` is not aligned with `def caveats; <<~EOS`.
Error: 2 problems in 1 formula
```

**Comment by jacobwgillespie** [comment]
Ha, this PR has been running long enough that the `audit` output keeps changing. 😛 

Updated.

**Comment by fxcoudert** [comment]
```
==> /usr/local/Cellar/docker-machine/0.15.0/bin/docker-machine create --driver hyperkit -h
Error: docker-machine-driver-hyperkit: failed
No such file or directory - /usr/local/Cellar/docker-machine/0.15.0/bin/docker-machine
```

**Comment by jacobwgillespie** [comment]
That's probably because it's missing `depends_on "docker-machine" => :recommended` like all the other `docker-machine-*`, but `audit` complains about recommended dependencies.  Should I add that back?

**Comment by DomT4** [comment]
What's the risk that people complain if we make it mandatory rather than `:recommended`?

**Review by mys721tx — COMMENTED** [review]
The test failed due to missing docker-machine. docker-machine should be added as one of the dependencies. Or the test case should be modified so that it would not depend on docker-machine.

**Comment by DomT4** [comment]
Merged in https://github.com/Homebrew/homebrew-core/commit/6108b2b183404663f720b4b8cd8e1d4efeaf9673. Thank you for your contribution to Homebrew @jacobwgillespie, and the patience in waiting for review; we appreciate both! :smiley_cat:

**Comment by jacobwgillespie** [comment]
Awesome, thank you!

**Pull request closed without merging**